### PR TITLE
Reports

### PR DESCRIPTION
**Pull request by christiangrothaus**

### CODE REVIEW

File: app/src/main/java/com/example/maintenancespace/ui/cars/CarSpinner.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+
+            }
+        });
+    }
+
+}
```
Comment (christiangrothaus) [inline app/src/main/java/com/example/maintenancespace/ui/cars/CarSpinner.java:59]: We can use this in the forms once this is merged into main for selecting cars.

File: app/src/main/java/com/example/maintenancespace/models/cars/CarModel.java [hunk 2]
```diff
@@ -68,4 +68,9 @@ public String getOwnerId(){
     public ArrayList<String> getUserIds(){
         return userIds;
     }
+
+    @Override
+    public String toString() {
+        return this.getYear() + " " + this.getMake() + " " + this.getModel() + " " + this.getTrim();
+    }
```
Comment (christiangrothaus) [inline app/src/main/java/com/example/maintenancespace/models/cars/CarModel.java:75]: Needed for the car spinner.

File: app/src/main/res/xml/provider_paths.xml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<paths>
+    <files-path name="files" path="."/>
+</paths>
```
Comment (christiangrothaus) [inline app/src/main/res/xml/provider_paths.xml:4]: Needed for accessing the app files.

File: app/src/test/java/com/example/maintenancespace/MaintenanceEventUnitTest.java [hunk 4]
```diff
@@ -1,17 +0,0 @@
...
-import org.junit.Test;
-
-public class MaintenanceEventUnitTest {
-    @Test
-    public void shouldGetCorrectName() {
-        MaintenanceEventModel event = new MaintenanceEventModel("Oil Change", "Description", new Timestamp(100,100));
-        String expected = "Oil Change";
-        assertEquals(event.getName(), expected);
-    }
-}
```
Comment (christiangrothaus) [inline app/src/test/java/com/example/maintenancespace/MaintenanceEventUnitTest.java:17]: Renamed this file.

File: app/src/main/java/com/example/maintenancespace/utilities/CsvWriter.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        try {
+            // Create a new intent for creating a file
+            Intent intent = new Intent(Intent.ACTION_CREATE_DOCUMENT);
+            intent.addCategory(Intent.CATEGORY_OPENABLE);
+            intent.putExtra(Intent.EXTRA_TITLE, "events.csv");
+            intent.setType("text/csv");
+            intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+
+            // Create a holding file for the data
+            File eventsFile = new File(activity.getFilesDir(), "events.csv");
```
Comment (christiangrothaus) [inline app/src/main/java/com/example/maintenancespace/utilities/CsvWriter.java:26]: The `getFilesDir` method is making use of that provider in the `provider_paths.xml` file.

File: app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -11,7 +11,7 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.MaintenanceSpace"
-        tools:targetApi="31">
```
Comment (christiangrothaus) [inline app/src/main/AndroidManifest.xml:14]: This was set to the wrong API version.

**Review by gaboonsk — DISMISSED** [review]

**Review by gaboonsk — APPROVED** [review]

**Review by gaboonsk — APPROVED** [review]

**Pull request merged**